src/platform/qt/library/LibraryModel.cpp

```diff
@@ -0,0 +1,424 @@
+/* Copyright (c) 2013-2022 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "LibraryModel.h"
+
+#include "../utils.h"
+
+#include <QApplication>
+#include <QDir>
+#include <QItemSelectionModel>
+#include <QSortFilterProxyModel>
+#include <QStyle>
+
+#include <algorithm>
+
+using namespace QGBA;
+
+LibraryModel::LibraryModel(QObject* parent)
+	: QAbstractItemModel(parent)
+	, m_treeMode(false)
+	, m_showFilename(false)
+{
+	QIcon iconGBA;
+	iconGBA.addFile(":/res/gba-icon-256.png", QSize(256, 256));
+	iconGBA.addFile(":/res/gba-icon-128.png");
+
+	QIcon iconGBC;
+	iconGBC.addFile(":/res/gbc-icon-128.png", QSize(128, 128));
+	iconGBC.addFile(":/res/gbc-icon-256.png", QSize(256, 256));
+
+	QIcon iconGB;
+	iconGB.addFile(":/res/gb-icon-128.png", QSize(128, 128));
+	iconGB.addFile(":/res/gb-icon-256.png", QSize(256, 256));
```

Comment:
I've rendered some more sizes, but they look kinda crappy since the icon was designed to be viewed at 128x128. I'm probably gonna upload some specific sizes though; I don't want to add several dozen kB to the repo that may or may not be useful.